tests/fixtures/invalid-template-attributes/basic/expected/x/component/component.html

```diff
@@ -0,0 +1,5 @@
+<template lwc:render-mode="light">
+    <div>
+        <template>click me!</template>
```

Comment:
How helpful is this exactly? AFAICT this codemod would turn a warning into an error. Before, you get a warning about the event listener, but after, you get an [error](https://stackblitz.com/edit/salesforce-lwc-uxgheg?file=src%2Fmodules%2Fx%2Fapp%2Fapp.html&title=LWC%20playground) because of the bare `<template>`:

```
Error: LWC1077: Invalid template tag. A directive is expected to be associated with the template tag.
```

IMO this codemod should only remove the `onclick` in a case where it actually helps elucidate LWC's real behavior, e.g.:

```html
<template for:each={rows} for:item="row" onclick={onClick}>
    <span key={row}></span>
</template>
```

In the above case, the codemod would remove the `onclick` and the result would be a valid template. Maybe in the other case, the codemod should throw an error or something?

Reply:
Hrm that's a good point, it will convert the warning to an error.

However, the compiled output if we don't remove the attributes from the `<template>` (when it's a warning), doesn't render anything.

For [example](https://stackblitz.com/edit/salesforce-lwc-zstj2b?file=src%2Fmodules%2Fx%2Fapp%2Fapp.html):

```html
<template>
    <template class="slds">
        <span>howdy</span>
    </template>
</template>
```

Doesn't render any of the HTML in the `<template>` tag.

For the codemod to be useful in this specific scenario I feel like we should convert the `<template>` to another tag (maybe `<div>`?) so the content will render as intended.

This is with the assumption that the component author realizes the content in the `<template>` without valid template directives doesn't actually render anything.

We could also add some notes in the codemod to let the component author know what's going to happen.

> the codemod should throw an error or something?

We could also throw an error/warning here too, I think the advantage of doing it here rather than in the LWC compiler is the component author is paying attention to fixing the HTML.

Let me know which option you would prefer.

Reply:
> I feel like we should convert the <template> to another tag (maybe <div>?)

Can we remove it entirely? This seems closer to the spirit of what LWC is doing under the hood. Adding a `<div>` may mess with styles (i.e. `display: block`).

Now that I think about it, throwing an error might not be great, because I'm not sure how that would work in the context of an IDE. Whereas removing the `<template>` entirely (i.e. doing whatever LWC actually does at runtime) does a great job of communicating the issue IMO.

Reply:
Sounds good to me!